example_7/controller/r6bot_controller.cpp

```diff
@@ -160,14 +164,30 @@ controller_interface::return_type RobotController::update(
 
   if (trajectory_msg_ != nullptr)
   {
-    interpolate_trajectory_point(*trajectory_msg_, time - start_time_, point_interp_);
+    bool reached_end;
```

Comment:
I assume it's not that terrible to allocate a bool in a controller loop, but I can make this a member variable if preferred.

Reply:
IMHO this is fine